pandas/core/internals/managers.py

```diff
@@ -1362,8 +1364,11 @@ def column_setitem(self, loc: int, idx: int | slice | np.ndarray, value) -> None
         # this manager is only created temporarily to mutate the values in place
         # so don't track references, otherwise the `setitem` would perform CoW again
         col_mgr = self.iget(loc, track_ref=False)
-        new_mgr = col_mgr.setitem((idx,), value)
-        self.iset(loc, new_mgr._block.values, inplace=True)
+        if inplace:
+            col_mgr.setitem_inplace(idx, value)
```

Comment:
does idx vs (idx,) matter here?  i did _almost_ the same thing in #49775 but used the tuple

Reply:
Since it's a method specific for SingleBlockManager (and what you do under the hood is `self.array[idx] = ..` (and so self.array being 1D), I think it shouldn't matter. 

Also originally before https://github.com/pandas-dev/pandas/pull/47074, in `DataFrame._set_value`, it was called without putting in a tuple